var arr = {1,2,3,4};
         # 0 1 2 3
try do
    print arr[0];
    print arr[1];
    print arr[2];
    print arr[3];
    print arr[4];
catch IndexException do
    print "IndexException!!!";
end_try

var x = 1;
var y = 1000;

var result = x + y;

print x;
print "RESULT: ", result;

func print_x()
    print "PRINTED: ", x;
end_func

print_x();

for i=(1,5) do
    print loop_i, x;
end_loop

func test_2()
    for i=(1,5) do
        print_x();
    end_loop
end_func

test_2();

func test(x, y, z)
    for i=(1,5) do
        try do
            var x = 2;
        catch ObjectException do
            try do
                #Тут намеренная ошибка синтаксиса
                print "ObjectException 1", x
            catch SyntaxException do
                print "SyntaxException 2";
            end_try
        end_try
        print "Hello!", x, loop_i;

        print "x: ", func_x, "y: ", func_y, "z: ", func_z;
    end_loop

    print "END_LOOP";

    try do
        print loop_i;
    catch ObjectException do
        print "ObjectException 3";
    end_try

    print "END_TRY1";

    print "x: ", func_x, "y: ", func_y, "z: ", func_z, "END";
    print "END FUNC";
end_func

test(1, 2, 3);

func test_wrapper_loops()
    for i=(1,4) do
        for j=(1,2) do
            for z=(1,23) do
                for k=(1,2) do
                    print loop_i, loop_j, loop_z, loop_k;

                    for tim=(-1,2) do
                        print "Tim: ", loop_tim;
                    end_loop

                    try do
                        :LSDKFH;lkh
                    catch SyntaxException do
                        print "SyntaxException 1";
                    end_try
                end_loop
            end_loop
        end_loop
    end_loop
end_func


test_wrapper_loops();
